rust/src/nfs/nfs3.rs

```diff
@@ -320,9 +320,9 @@ impl NFS3State {
 
     pub fn new_tx(&mut self) -> NFS3Transaction {
         let mut tx = NFS3Transaction::new();
-        self.tx_id += 1;
         tx.id = self.tx_id;
-        return tx;
+        self.tx_id += 1;
```

Comment:
I don't think this is correct, if you look at the lookup code in get_tx_by_id you can see it accounts for the off by one: if tx.id == tx_id + 1 {